openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
 class YarnStatusGetter(JobMetadataGetterInterface):
     """YARN app status getter"""
 
+    def get_application_url(self, application_id: str) -> str:
+        return url_join(
+            ConfigParams().yarn_rest_api_base_url,
+            f"/ws/v1/cluster/apps/{application_id}",
+        )
+
+    def get_authentication_provider(self):
+        auth = None
+        if ConfigParams().yarn_auth_use_kerberos:
+            # TODO: HTTPKerberosAuth is deprecated, this is a utility class for compatibility with requests_kerberos.
+            #   Find the better way to do this, using requests_gssapi directly.
+            # Have to set mutual_authentication to REQUIRED or it just skips kerberos auth it seems.
+            auth = requests_gssapi.HTTPKerberosAuth(
```

Comment:
Is it possible to also address this TODO? `HTTPSPNEGOAuth` seems to be the replacement.

Reply:
Done: now its using requests_gssapi.HTTPSPNEGOAuth directly, and only where we really need to use authentication.
(So not in the unit tests)